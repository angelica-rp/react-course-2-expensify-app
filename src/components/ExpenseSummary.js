import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';


export const ExpenseSummary = (props) => {
    const expenseWord = props.totalCount === 1 ? 'expense' : 'expenses' ;
    const formattedExpensesTotal = numeral(props.totalAmount / 100).format('$0,000.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{props.totalCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal} </span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>    
            </div>
        </div>
    );
};
 
const mapStateToProps = (state) => {
    const expenses= getVisibleExpenses(state.expenses, state.filters);
    return {
        totalCount: expenses.length,
        totalAmount: getExpensesTotal(expenses)
    }
}; 

export default connect(mapStateToProps)(ExpenseSummary);
