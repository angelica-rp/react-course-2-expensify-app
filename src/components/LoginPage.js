import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle, startLoginFacebook } from '../actions/auth';

export const LoginPage = ({ startLoginGoogle, startLoginFacebook }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify Application</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button button--login" onClick={startLoginGoogle}>Login with Google</button>
            <button className="button button--login" onClick={startLoginFacebook}>Login with Facebook</button>
        </div>
    </div>    
);

const mapDispatchToProps = (dispatch) => ({
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginFacebook: () => dispatch(startLoginFacebook())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);