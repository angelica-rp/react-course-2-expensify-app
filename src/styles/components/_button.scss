.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--login {
    width: 80%;
    margin: $s-size;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}