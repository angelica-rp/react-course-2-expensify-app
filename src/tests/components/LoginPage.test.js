import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/loginpage';

test('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLoginGoogle on button click', () => {
    let startLoginGoogle = jest.fn();
    let wrapper = shallow(<LoginPage startLoginGoogle={startLoginGoogle} />);
    wrapper.find('button').at(0).simulate('click');
    expect(startLoginGoogle).toHaveBeenCalled();
});