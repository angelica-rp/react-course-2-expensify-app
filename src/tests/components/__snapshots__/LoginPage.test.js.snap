// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify Application
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <button
      className="button button--login"
    >
      Login with Google
    </button>
    <button
      className="button button--login"
    >
      Login with Facebook
    </button>
  </div>
</div>
`;
